src/components/AccountForms/RegistrationForm/index.js

```diff
@@ -202,40 +201,27 @@ const RegistrationForm = ({ mockOnSubmit }) => {
     retypePassword: false,
   });
   const history = useHistory();
-  // const [error, setError] = useState({ userName: { message: '' } });
 
-  /*const key = Object.keys(error || {}).toString();
-  if (key === 'email') {
-    const { email } = error;
-    errors.email = email;
-  }*/
+  const onSubmit = async (data) => {
+    try {
+      await request.post(API_PATH.REGISTER, data);
 
-  const onSubmit = (data) => {
-    request
-      .post(API_PATH.REGISTER, data)
-      .then(({ status }) => {
-        if (status === 200) {
-          const inProd = process.env.NODE_ENV === 'production';
-          const options = {
-            category: 'onSubmit',
-            action: 'Created an Account',
-          };
-          inProd && event(options);
-          dispatch(updateAuth(auth.checkCookie()));
-          history.push('/mentors');
-        }
-      })
-      .catch(({ err }) => {
-        const message = _get(
-          err,
-          `${Object.keys(err).toString()}.message`,
-          'Registration error occured'
-        );
-        setError(Object.keys(err).toString(), {
-          type: 'manual',
-          message,
-        });
+      const inProd = process.env.NODE_ENV === 'production';
```

Comment:
on second thought, wouldn't it be more intuitive if we default to prod and make the other environments the exception?

Reply:
we only need google analytics in prod, and it's giving us an error in the unit test if GA is not skipped in dev environment